dashboard/app/views/shared/_user_stats.html.haml

```diff
@@ -19,8 +19,9 @@
       .games
         - last_level_id = nil
         - level_repeat = 0
-        - stage[:levels].each do |script_level|
-          - level_id = level_with_best_progress(script_level[:ids], level_progress)
+        - stage[:script_levels].each do |script_level|
+          - level_id = level_with_best_progress(script_level[:levels].map{|l| l[:id]}, level_progress)
+          - level = script_level[:levels].find {|l| l[:id] == level_id}
```

Comment:
I'm not sure that this partial has any/very good test coverage, so it might be worth some manual validation here if you haven't done so already.